### PR TITLE
dotCMS/core#16943 Paginator in form selection dialog is blocking last row

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.html [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+    [contentStyle]="{ padding: '0' }"
+    [cssClass]="paginatorService.totalRecords > paginatorService.paginationPerPage ? 'paginator' : ''"
+    (hide)="close.emit($event)"
+    [header]="messages['modes.Add-Form']"
+>
     <p-dataTable
+        #datatable
         (onLazyLoad)="loadData($event)"
         [lazy]="true"
+        [style]="paginatorService.totalRecords > paginatorService.paginationPerPage ? { 'min-height': contentMinHeight + 'px' } : ''"
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.html]: maybe we should move this 

```
paginatorService.totalRecords > paginatorService.paginationPerPage
```

to a method

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.html [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
-        [paginator]="true"
+        [paginator]="paginatorService.totalRecords > paginatorService.paginationPerPage"
         [rows]="paginatorService.paginationPerPage"
         [totalRecords]="paginatorService.totalRecords"
-        [value]="items">
+        [value]="items"
+    >
         <p-column [header]="messages['contenttypes.form.name']" field="name"></p-column>
-        <p-column [style]="{'text-align': 'right'}">
+        <p-column [style]="{ 'text-align': 'right' }">
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.html]: should it be inline?

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts [hunk 3]
```diff
@@ -39,8 +56,17 @@ export class DotFormSelectorComponent implements OnInit {
...
             });
 
+        this.paginatorService.paginationPerPage = 5;
         this.paginatorService.url = 'v1/contenttype?type=FORM';
     }
+    ngOnChanges(changes: SimpleChanges) {
+        setTimeout(() => {
+            if (changes.show.currentValue === true) {
+                this.contentMinHeight =
+                    this.dialog.dialog.nativeElement.getBoundingClientRect().height - 100;
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts]: dialog.dialog?

maybe we should change this one

https://github.com/dotCMS/core-web/pull/1108/files#diff-0db0fb48a9fe56dc260a60ff6b57db62R38

for something like: 'dotDialogComponent'

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.html [hunk 4]
```diff
@@ -1,21 +1,32 @@
...
                     pButton
                     flat
                     class="form-selector__button"
                     [label]="messages['Select']"
-                    (click)="select.emit(item)"></button>
+                    (click)="select.emit(item)"
+                ></button>
             </ng-template>
         </p-column>
     </p-dataTable>
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.html]: test?

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts [hunk 5]
```diff
@@ -39,8 +56,17 @@ export class DotFormSelectorComponent implements OnInit {
...
+            }
+        }, 0);
+    }
 
     /**
      * Call when click on any pagination link
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts]: test?

File: src/app/portlets/dot-edit-page/layout/components/dot-theme-selector/dot-theme-selector.component.html [hunk 6]
```diff
@@ -14,7 +15,7 @@
         <p-dataGrid
             #dataGrid
             [value]="themes | async"
-            [paginator]="true"
+            [paginator]="paginatorService.totalRecords > paginatorService.paginationPerPage"
             [rows]="paginatorService?.paginationPerPage"
             [totalRecords]="paginatorService?.totalRecords"
             [lazy]="true" (onLazyLoad)="paginate($event)"
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/layout/components/dot-theme-selector/dot-theme-selector.component.html]: test?

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts (hunk 3 above)
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts]: If you get the `height` from the `.ui-datatable` you don't need to substract anything and the problem with substract is that we change something in the header, then the math will break.

```suggestion
                    this.dialog.dialog.nativeElement.querySelector('.ui-datatable').getBoundingClientRect().height;
```

Comment (fmontes) [reply src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts]: maybe `#dotDialog`

File: src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts (hunk 5 above)
Comment (alfredo-dotcms) [reply src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.ts]: this is the component that has the tests `disabled`
https://github.com/dotCMS/core-web/blob/448b543f7ed45db45585b097689a67be111ef3e6/src/app/portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.component.spec.ts

for some reason when you enable it, fails completely

**Pull request merged**